SIM-ONE Training/prioritary_mvlm/enhanced_trainer.py

```
@@ -828,7 +832,30 @@ def train(self, num_epochs: Optional[int] = None) -> Path:
+                self.patience_counter = 0
+
+                # Save best model
+                self.save_checkpoint("best_model.pt")
+                self.logger.info(f"💾 New best model saved! Loss: {current_loss:.4f}")
+            else:
+                self.patience_counter += 1
+                self.logger.info(f"⏳ No improvement for {self.patience_counter} epoch(s). Best: {self.best_loss:.4f} at epoch {self.best_epoch}")
+
+            # Early stopping check (only after minimum epochs)
+            if (epoch_idx + 1) >= self.config.min_epochs and self.patience_counter >= self.config.patience:
+                self.logger.info(f"🛑 Early stopping triggered!")
```

Ruff (0.13.1)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)